settings.go

```diff
@@ -39,8 +41,26 @@ func extensionsDir() (string, error) {
 	switch runtime.GOOS {
 	case "linux", "darwin":
 		path = os.ExpandEnv("$HOME/.vscode/extensions/")
+	case "windows":
+		return os.ExpandEnv("/c/Users/$USERNAME/.vscode/extensions"), nil
 	default:
 		return "", xerrors.Errorf("unsupported platform: %s", runtime.GOOS)
 	}
 	return filepath.Clean(path), nil
 }
+
+func windowsVarFix(input string) string {
```

Comment:
This function seems unnecessary, I think it should be removed and my suggestions on it's callers be implemented.

Reply:
Is needed for now unless you know a better way that does not involve VarShadowing (yes, i tried your method, did not work as it did var shadowing for me), also if i recall correctly, there are a few places where i need to place this, so wouldnt it be better to just, declare it once?

Reply:
and yes, i am aware the function looks horrible, it was actually the first function i wrote from the ground up without copying an example off the internet and tweaking it

Reply:
It's used in two places, and it's unclear what it does based on the name. It also doesn't verify that the input doesn't already have `~/`.

The callers should be changed to the following to make it more clear what is going on in the code. If there were more than two callers or the function was bigger I would encourage having a helper function but in this case it seems like it adds more confusion than it saves.

```go
whatever := "~/.local/blah"
if runtime.GOOS == "windows" {
        whatever = ".local/blah"
}

Reply:
Will clean up when I get a chance

Reply:
Added a related comment in one of the locations using this function, hope you guys don't mind the unsolicited review.

Reply:
No worries, its good that someone took a look.